tests/sentry/monitors/consumers/test_monitor_consumer.py

```diff
@@ -396,6 +396,27 @@ def test_check_in_update_terminal(self):
         checkin = MonitorCheckIn.objects.get(guid=self.guid)
         assert checkin.duration == int(20.0 * 1000)
 
+    def test_check_in_update_terminal_user_terminal(self):
+        monitor = self._create_monitor(slug="my-monitor")
+        self.send_checkin(monitor.slug, duration=10.0)
+        self.send_checkin(monitor.slug, guid=self.guid, status="in_progress")
+
+        checkin = MonitorCheckIn.objects.get(guid=self.guid)
+        assert checkin.duration == int(10.0 * 1000)
+
+        self.send_checkin(monitor.slug, duration=20.0, status="error")
+        self.send_checkin(
+            monitor.slug,
+            guid=self.guid,
+            status="ok",
+            expected_error=ProcessingErrorsException(
+                [{"type": ProcessingErrorType.CHECKIN_FINISHED}], monitor
+            ),
+        )
+
+        checkin = MonitorCheckIn.objects.get(guid=self.guid)
+        assert checkin.duration == int(20.0 * 1000)
```

Comment:
Does this actually validate that we didn't create a processing error?

Reply:
nope, unfortunately we don't have validation for when we don't create a processing error. Right now I've just opted to test that we actually do create an error for the case where it should

Reply:
Can we have it check that no errors were created if `expected_error` is some sentinel value?

Reply:
Yea right now expected_error is default None, but that just ignores checking for any sort of expected_error. I'm planning on changing the None to validate that no processing_errors were created, but that also involves adding an actual expected_error to many of the tests in this file so I was planning on doing in a follow-up.

Though actually, maybe I can work around this for now 1 sec